Enable caching of catkin_pkg.packages.find_packages when using roslaunch

garaemon opened the pull request:
patch for https://github.com/ros/ros_comm/issues/541

garaemon committed:
Enable caching of catkin_pkg.packages.find_packages when using roslaunch

ros-pull-request-builder commented:
Can one of the admins verify this patch?

dirk-thomas commented:
Please consider to update the PR according to the comment made on the related PR https://github.com/ros/catkin/pull/704#issuecomment-68476827

garaemon committed:
Use source_path_to_packages keyword argument of
catkin.find_in_workspaces.find_in_workspaces to cache packages in catkin
workspaces and add _catkin_packages_cache global variables to several modules

dirk-thomas commented on tools/roslaunch/src/roslaunch/__init__.py:
```diff
@@ -203,7 +204,13 @@ def main(argv=sys.argv):
+        global catkin_packages_cache
+        catkin_packages_cache = roslib.packages.catkin_packages_cache()
+        # set cache for each modules
+        rlutil._catkin_packages_cache = catkin_packages_cache
```
Accessing private variables is not recommended.

dirk-thomas commented on tools/roslaunch/src/roslaunch/node_args.py:
```diff
@@ -56,6 +56,8 @@
 from roslaunch.config import load_config_default
 import roslaunch.xmlloader
 
+_catkin_packages_cache = None
```
Why do multiple files have this variable?

wkentaro commented:
@dirk-thomas 
I sent PR with another implementation, so please review:
- https://github.com/ros/ros_comm/pull/676

And it depends:
- https://github.com/ros/ros/pull/91
- https://github.com/ros/catkin/pull/757
